src/net-utils.ts

```diff
@@ -39,6 +39,63 @@ async function isPortOpen(
   });
 }
 
+/**
+ * Repeatedly attempts to connect to a specified port until it becomes available
+ * or times out. Uses a shorter connection timeout (200ms) for each attempt
+ * while respecting the overall timeout period.
```

Comment:
This is backwards, isn't it? It doesn't try to connect to the given port, but rather, listens?

Also, I think the time between attempts is constant.

Reply:
Good catch -- I wrote these docs for a prior version of the code. I'll update.